packages/devtools_app/lib/src/screens/memory/allocation_profile_table_view.dart

```diff
@@ -0,0 +1,424 @@
+
+  late AllocationProfileTableViewController allocationProfileController;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (!initController()) {
+      return;
+    }
+    allocationProfileController = controller.allocationProfileController
+      ..initialize();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        _AllocationProfileTableControls(
+          allocationProfileController: allocationProfileController,
+          refreshOnGcKey: refreshOnGcKey,
+          refreshKey: refreshKey,
+        ),
+        const SizedBox(
+          height: denseRowSpacing,
+        ),
+        Expanded(
+          child: _AllocationProfileTable(
+            allocationProfileController: allocationProfileController,
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _AllocationProfileTable extends StatelessWidget {
+  const _AllocationProfileTable({
+    Key? key,
+    required this.allocationProfileController,
+  }) : super(key: key);
+
+  /// List of columns that are displayed regardless of VM developer mode state.
+  static final _columnGroups = [
+    ColumnGroup(
+      title: '',
+      range: const Range(0, 1),
+    ),
+    ColumnGroup(
+      title: 'Total',
+      range: const Range(1, 5),
+    ),
+  ];
+
+  static final _columns = [
+    FieldClassName(),
+    FieldInstanceCountColumn(heap: HeapGeneration.total),
+    FieldSizeColumn(heap: HeapGeneration.total),
+    FieldInternalSizeColumn(heap: HeapGeneration.total),
+    FieldExternalSizeColumn(heap: HeapGeneration.total),
+  ];
+
+  /// Columns for data that is only displayed with VM developer mode enabled.
+  static final _vmDeveloperModeColumnGroups = [
+    ColumnGroup(
+      title: 'New Space',
+      range: const Range(5, 9),
+    ),
+    ColumnGroup(
+      title: 'Old Space',
+      range: const Range(9, 13),
+    ),
+  ];
+
+  static final _vmDeveloperModeColumns = [
+    FieldInstanceCountColumn(heap: HeapGeneration.newSpace),
+    FieldSizeColumn(heap: HeapGeneration.newSpace),
+    FieldInternalSizeColumn(heap: HeapGeneration.newSpace),
+    FieldExternalSizeColumn(heap: HeapGeneration.newSpace),
+    FieldInstanceCountColumn(heap: HeapGeneration.oldSpace),
+    FieldSizeColumn(heap: HeapGeneration.oldSpace),
+    FieldInternalSizeColumn(heap: HeapGeneration.oldSpace),
+    FieldExternalSizeColumn(heap: HeapGeneration.oldSpace),
+  ];
+
+  final AllocationProfileTableViewController allocationProfileController;
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<AllocationProfile?>(
+      valueListenable: allocationProfileController.currentAllocationProfile,
+      builder: (context, profile, _) {
+        // TODO(bkonyi): make this an overlay so the table doesn't
+        // disappear when we're retrieving new data, especially since the
+        // spinner animation freezes when retrieving the profile.
```

Comment:
is there a way to split up the processing into batches or use another isolate so that we can release the UI thread for the spinner?

Reply:
As far as I know, no. `getAllocationProfile` is atomic and can't be broken down, and I'm pretty sure this is what's causing the wait. Web also doesn't support isolates (unless there's another option for spinning up another thread that I'm not aware of).

Maybe this can be fixed by removing some `await`s and using callbacks to notify the UI when the request is complete.